src/cudnn/tensor/descriptor.rs

```diff
@@ -0,0 +1,66 @@
+use super::super::sys::*;
+use crate::prelude::*;
+use core::marker::PhantomData;
+use core::mem::MaybeUninit;
+
+pub(crate) const TENSOR_FORMAT: cudnnTensorFormat_t = cudnnTensorFormat_t::CUDNN_TENSOR_NCHW;
+/// A descriptor of a tensor. This can be reused as all tensors are fully packed
+/// and in NCHW layout.
+///
+/// This descriptor is destroyed when it is dropped.
+///
+/// # See also
+/// <https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnTensorDescriptor_t>
+/// <https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnDestroyTensorDescriptor>
+pub struct TensorDescriptor<T, const N: usize, const C: usize, const H: usize, const W: usize> {
```

Comment:
Yeah this can be changed to hold runtime values instead of const generics

Reply:
ok, but should this runtime panic (asserts) or just be unsafe?

Reply:
(this is most likely affecting every module in cudnn 😅)

Reply:
runtime panics on creation IMO